core/src/main/java/com/sequenceiq/cloudbreak/service/environment/credential/CredentialClientService.java

```diff
@@ -60,4 +62,16 @@ public Credential getByEnvironmentCrn(String envCrn) {
             throw new CloudbreakServiceException(message, e);
         }
     }
+
+    @Override
+    public String getCloudPlatformByCredential(String credentialName, String accountId, CredentialType type) {
+        try {
+            CredentialResponse credentialResponse = credentialEndpoint.getByName(credentialName);
+            return credentialResponse.getCloudPlatform();
+        } catch (WebApplicationException | ProcessingException | IllegalStateException e) {
```

Comment:
we usually handle `WebApplicationException` separately to extract the real reason

Reply:
i have removed it